go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -94,13 +137,35 @@ func deleteService(service *kapi.Service, inport, gwBridge string, nodeIP *net.I
 					"%s, stderr: %q, error: %v", gwBridge, externalIP, stderr, err)
 			}
 		}
+		for _, ing := range service.Status.LoadBalancer.Ingress {
+			if ing.IP == "" {
+				continue
+			}
+			ingIP := net.ParseIP(ing.IP)
+			if ingIP == nil {
+				klog.Errorf("Failed to parse ingress IP: %s", ing.IP)
+				continue
+			}
+			cookie, err := svcToCookie(service.Name, ingIP, svcPort.Port)
+			if err != nil {
+				klog.Errorf("Unable to generate cookie for svc: %s, %s, %d, error: %v",
+					service.Name, ingIP.String(), svcPort.Port, err)
+			}
+			_, stderr, err := util.RunOVSOfctl("del-flows", gwBridge,
+				fmt.Sprintf("cookie=%s/32", cookie))
```

Comment:
Given the way this code currently works (where Updates are handled as Delete+Add) you could just have a single cookie per Service rather than having it depend on the IP and port too, and then you just need a single del-flows call when deleting the Service. (In that case you'd have to make `svcToCookie` take the service namespace too, since just "name" is not globally unique. [In theory it should take the namespace now too, but actually just ingressIP+port is globally unique, so it really doesn't even need the name at all...])

But depending on your future refactoring plans, the current code may make more sense.

Reply:
yeah, I was trying to make the cookie as unique as possible given our discussion earlier today about future refactoring